### ISSUE TITLE
Too many cf_bypass instances when running more than 1 task at the same time

### ISSUE THREAD

**Issue by spamtaztik** [fmd-project-team/FMD#197]
I don't know if it's just me, but it seems like checking for favorites in 9.164 became quite a bit longer than usual. When I checked the task manager I saw that fmd called a bunch of cf_bypass.exe's from time to time, which will then disappear on their own after a while.

Now, the problem:
1. Whenever cf_bypass is running, it seems like FMD's favorite checking speed goes down
2. FMD calls cf_bypass quite a number of times, making favorite checking even slower
3. My favorite list contains mangas from quite a few different sites, and the only ones that need the CF workaround are only Mangadex and Kissmanga (heck, earlier today I still need to change the cookie for Kissmanga after updating to 9.164). I've never did the CF workaround for sites other than MD and Kiss.

TL;DR: is it possible to add a feature (a checklist, or maybe in options > websites > advanced) so that fmd will run cf_bypass only when checking from those two sites?

Much obliged.

EDIT: sorry if this is a stupid question. I'm not a web programmer.

**Comment by AbidZakaria** [comment]
I don't know whether this works for you or not
Favorite manga checking is made not automatic
exit the program and open again
go to favorite tab and check manually 5 manga
if you have tried checking 5 manga down, just don't go straight to 10 manga later the program not responding
if you have done it all, please check all the mages guaranteed to be 50% successful

**Comment by SDXC** [comment]
@spamtaztik Please zip your FMD installation and upload it in here. Either cf_bypass has a problem with the website, a setting of your FMD installation is the problem or the filesystem blocks denies access to the temp file created in the config folder.

**Comment by spamtaztik** [comment]
@SDXC Should I include the data folder as well?

**Comment by spamtaztik** [comment]
My fmd folder zipped is 30 megs and github won't let me. Any other way to upload? 

Any specific files/folders you need (so the zipped file won't be as big)?

**Comment by SDXC** [comment]
@spamtaztik I actually don't need the data folder. Your works and config folder are more important.

It's not about a setting you actively set after the update, but more about a setting you set that the cf_bypass can't handle right now, so I want to find out which it is.

**Comment by spamtaztik** [comment]
[Free Manga Downloader.zip](https://github.com/fmd-project-team/FMD/files/3212784/Free.Manga.Downloader.zip)

Works and config? You got it. Do let me know if you need anything else.

**Comment by SDXC** [comment]
@spamtaztik You running the x86 or x64 version of FMD?

**Comment by spamtaztik** [comment]
@SDXC back when I downloaded the forked version (9.163) I chose the 64 bit version. I'm assuming the 9.164 one is also 64 bit... 

@AbidZakaria I have no problem not being able to check the favorites. The check runs until the end, but much slower than usual. Of course manual check works like a charm.

**Comment by SDXC** [comment]
Well okay, I think I can see where the problem lies.
You have 3 tasks for the favorite checks and cf_bypass has problems with more than 1 task it seems. The problem starts probably when 2-3 different websites need to run cf_bypass at the same time and they get into a conflict, resulting into the values being overwritten or deleted too early, etc.

A workaround could be to set "Number of downloaded files per task at the same time" to "1".
It may be slower but should at least be more stable.

**Comment by Tmp341** [comment]
CF doesn't like too many requests.

**Comment by ChocolateOtaku** [comment]
I think the major problem is that FMD shares the output from external processes via a file instead of STDOUT stream.
How can you assure the content of the file belongs to the sender if the content may be overwritten...

**Comment by SDXC** [comment]
Not sure why, but either Python or PyInstaller don't or at least not correctly write to STDOUT.
It was always empty when trying to get the STDOUT content.

That's why I went for the file exchange, of course only with 1 task at a the time which also ensures that the website is always correct. With more tasks it's of course not correct anymore at some point.

It's a really rough approach with much room for improvement.

**Comment by ChocolateOtaku** [comment]
In my opinion you are limiting the capabilities of NodeJS by wrapping it in Python instead of directly utilizing NodeJS' full potential and those of all the available packages...

![Untitled Diagram](https://user-images.githubusercontent.com/38581381/58281475-8a016800-7da3-11e9-918c-faee3c453ee9.png)

**Comment by spamtaztik** [comment]
@SDXC So run the check 1 at a time? Okey dokey, I'll change it then. 

Is this a temporary thing until you managed to tweak some more things or permanent?
*please say temporary please please please*

**Comment by SDXC** [comment]
@spamtaztik It's temporary

**Comment by spamtaztik** [comment]
@SDXC Just letting you know that kissmanga required me to update their cookie again just now. Even with cf_bypass running and I'm only checking 1 favorites at a time.

**Comment by SDXC** [comment]
FMD 0.9.165.0 has been released.
